OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -237,13 +245,19 @@ void ITick.Tick(Actor self)
 
 		protected virtual void Tick(Actor self)
 		{
+			if (queue.Count > 0 && enabledProductionTraits.Count == 0)
```

Comment:
I somehow managed to get an NRE here with `queue.Count` being 1 and `enabledProductionTraits` being `null`, but `productionTraits` being an empty collection.
You should initialize `enabledProductionTraits` in `INotifyCreated.Created` as well to avoid that, and probably change this to `queue.Count > 0 && productionTraits.Count > 0 && enabledProductionTraits.Count == 0` to skip some queue clearing when `productionTraits` is empty (although I'm not sure how that's empty).
Also `ClearQueue` contains a `queue.Count` check, so that's redundant.

Reply:
> when productionTraits is empty (although I'm not sure how that's empty).

Could that be because it's a list and not an array or enumerable?